src/main/fc/config.c

```diff
@@ -447,6 +447,31 @@ static void validateAndFixConfig(void)
     }
 #endif
 
+#if defined(USE_INTEGRATED_YAW_CONTROL)
+    for (unsigned i = 0; i < PID_PROFILE_COUNT; i++) {
+        const pidProfile_t* pidProfile = pidProfiles(i);
+        pidProfile_t* mutableProfile = pidProfilesMutable(i);
+        if (pidProfile->use_integrated_yaw && pidProfile->pid[PID_YAW].I > 30) {
+            mutableProfile->pid[PID_YAW].P = 30;
```

Comment:
I'm not too sure about this way... I understand that people must use less I with integrated yaw, but reset all the values when a user has an I > 30 seems a little strange for users.

Reply:
I agree with you. I'm just trying to detect whether the user has pasted an invalid diff from earlier versions. High I would indicate that. We could also base it on D entirely. D is needed with integrated yaw, but doesn't work with normal yaw. Actually that might be better since not everybody might have picked up on the high I for traditional Yaw recommendation yet. If you agree I'll change it that way.

Reply:
I don't think there is a good way to do this. I only wanted to notice that but I don't like any of the solutions :)

Reply:
If USE_INTEGRATED_YAW_CONTROL is defined, it still could be turned off by the user.  In that case what PIDs would they get for yaw?
My feeling is that we should not add this level of complexity to the code.
If a user wants to enable integrated yaw they should read the relevant wiki entry, which would require a tuning guide, and choose appropriate settings for their quad, which could be provided in the wiki.

Reply:
It's not just a matter of choice - it's a matter of safety.  There is a learning curve that must be followed to enable Integrated Yaw, and an understanding of the significantly different P I D F values when Integrated Yaw is enabled.

Reply:
Joe has four holes in his bedroom ceiling due to having I gains at what we now think of as default for normal yaw - 130.  I don't think anything should be enabled that could result in a dangerous condition if someone blindly pastes a diff (as a lot of people do).